test/e2e/proxy_publicviewer_test.go

```diff
@@ -0,0 +1,313 @@
+						err = communityUserProxyClient.Create(context.TODO(), &cm)
+						require.True(t, errors.IsForbidden(err), "expected Create ConfigMap as community user to return a Forbidden error, actual: %v", err)
+					})
+				})
+			}
+		})
+
+		t.Run("user is banned", func(t *testing.T) {
+			for s, c := range tt {
+				t.Run(s, func(t *testing.T) {
+					user := c.proxyClientUser()
+					// the client needs to be created before the ban,
+					// otherwise it won't initialize properly
+					url := hostAwait.ProxyURLWithWorkspaceContext(sp.Name)
+					proxyClient, err := hostAwait.CreateAPIProxyClient(t, user.Token(), url)
+					require.NoError(t, err)
+
+					banUser(t, hostAwait, user)
+
+					t.Run(s, func(t *testing.T) {
+						t.Run("user can not initialize a new client", func(t *testing.T) {
+							url := hostAwait.ProxyURLWithWorkspaceContext(sp.Name)
+							proxyClient, err := hostAwait.CreateAPIProxyClient(t, user.Token(), url)
+							require.NoError(t, err)
+
+							// as the client is not initialized correctly,
+							// any request should return a NoMatch error
+							cm := corev1.ConfigMap{
+								ObjectMeta: metav1.ObjectMeta{
+									Name:      "test-cm",
+									Namespace: sp.Status.ProvisionedNamespaces[0].Name,
+								},
+							}
+							err = proxyClient.Create(context.TODO(), &cm)
+							require.True(t, meta.IsNoMatchError(err), "expected Create ConfigMap as SSO user to return a NoMatch error, actual: %v", err)
+						})
+
+						t.Run("cannot list config maps", func(t *testing.T) {
+							cms := corev1.ConfigMapList{}
+							err := proxyClient.List(context.TODO(), &cms, client.InNamespace(sp.Status.ProvisionedNamespaces[0].Name))
+							require.Zero(t, cms)
+							require.Error(t, err)
+						})
+
+						t.Run("cannot create config maps", func(t *testing.T) {
+							cm := corev1.ConfigMap{
+								ObjectMeta: metav1.ObjectMeta{
+									Name:      "test-cm",
+									Namespace: sp.Status.ProvisionedNamespaces[0].Name,
+								},
+							}
+							err := proxyClient.Create(context.TODO(), &cm)
+							require.True(t, errors.IsForbidden(err), "expected Create ConfigMap as SSO user to return a Forbidden error, actual: %v", err)
+						})
+					})
+				})
+			}
+		})
+	})
+
+	t.Run("space is not flagged as community", func(t *testing.T) {
+		// retrieve the space
+		sp := toolchainv1alpha1.Space{}
+		err := hostAwait.Client.Get(context.TODO(), client.ObjectKeyFromObject(space), &sp)
+		require.NoError(t, err)
+
+		// ensure no SpaceBinding exists for Public-Viewer
+		sbs, err := hostAwait.GetSpaceBindingByListing(toolchainv1alpha1.KubesawAuthenticatedUsername, space.Name)
+		require.NoError(t, err)
+		require.Empty(t, sbs)
+
+		testCases := map[string]bool{
+			"user is not banned": false,
+			"user is banned":     true,
+		}
+		for s, tc := range testCases {
+			t.Run(s, func(t *testing.T) {
+				for s, c := range tt {
+					t.Run(s, func(t *testing.T) {
+						user := c.proxyClientUser()
+						if tc {
```

Comment:
minor - maybe we can rename this to something more self explanatory like `banUser` , `isBanned` or something similar. Just to make it easier to read.

Reply:
done 9213645a5bbc6c7e2b3bebaa528d2abe05f5ecde